xyz-iinuwa-credential-manager-portal-gtk/src/credential_service/usb.rs

```diff
@@ -346,6 +443,32 @@ impl From<UsbStateInternal> for UsbState {
             UsbStateInternal::Completed(_) => UsbState::Completed,
             // UsbStateInternal::UserCancelled => UsbState:://UserCancelled,
             UsbStateInternal::SelectingDevice(_) => UsbState::SelectingDevice,
+            UsbStateInternal::SelectCredential { response, cred_tx } => {
+                UsbState::SelectCredential {
+                    creds: response
+                        .assertions
+                        .iter()
+                        .map(|x| Credential {
+                            id: x
```

Comment:
This will need to be used as a handle for the UI to return which credential the user selected. So this should return an error (unwrap is fine for now) rather than mapping to a non-unique string.

Separately, I think we should use an opaque ID here so that the credential ID is not linked to the (untrusted) UI code.

Reply:
Good catch regarding the `<unknown>`-fallback!
I also hashed the ID now, before sending it.